include/roboteam_utils/Line.h

```diff
@@ -1,207 +1,93 @@
- *
+ * The Line class represents an infinite length line, it is encoded by two different points through which the line is infinitely extended in both directions. If you want to use
+ * a finite length line instead then use LineSegment class instead.
+ * @author Created by: Rolf van der Hulst <br>
+ *         Documented and redesigned by: Haico Dorenbos
+ * @since 2019-04-18
  */
 class Line {
-    public:
-    /**
-     * @brief Start of the line
-     *
-     */
-    Vector2 start;
-    /**
-     * @brief End of the line
-     *
-     */
-    Vector2 end;
-    /**
-     * @brief Construct a new Line object
-     *
-     */
-    constexpr Line() = default;
-
-    /**
-     * @brief Constructs a new Line object
-     *
-     * @param _start Start of the Line
-     * @param _end End of the Line
-     *
-     */
-    constexpr Line(const Vector2 &_start, const Vector2 &_end) noexcept: start{_start}, end{_end} {};
-    /**
-     * @brief Constructs a new Line from a LineSegment.
-     * @param other LineSegment to use for construction
-     */
-    explicit Line(const LineSegment &other) noexcept;
-    /**
-     * @brief Gets the length of the vector representation of this line.
-     *        Note in this case it does not really make sense as this class represents lines of infinite length
-     * Literally:
-     *      (end - start).length()
-     * @return double Gets the length of the line
-     */
-    [[nodiscard]] double length() const;
-
-    /**
-     * @brief Gets the length of the vector representation of this line
-     *
-     * @return double Length of this Line
-     */
-    [[nodiscard]] double length2() const;
-
-    /**
-     * @brief Gets the slope of this line
-     *
-     * @return double Slope of the line
-     */
-    [[nodiscard]] double slope() const;
-
-    /**
-     * @brief Gets the intercept of this line
-     * Literally:
-     *      start.y - slope() * start.x;
-     * @return double Intercept of this line
-     */
-    [[nodiscard]] double intercept() const;
-
-    /**
-     * @brief Gets the direction of the Line
-     *
-     * @return Vector2 Vector representation of the direction of this vector
-     */
-    [[nodiscard]] Vector2 direction() const;
+   public:
+    Vector2 v1;  // One of the points located on the Line.
+    Vector2 v2;  // Another point located on the Line, which should be different than the v1.
 
     /**
-     * @brief Gets a pair of coefficients
+     * Creates a new Line instance by using 2 different points on that Line. Make sure that the given points are different.
      *
-     * @return std::pair<double, double> Pair of doubles where .first == slope() and .second == intercept()
+     * @param v1 An arbitrary point on the Line.
+     * @param v2 An arbitrary different point on the Line (make sure that it is different than the v2 point).
      */
-    [[nodiscard]] std::pair<double, double> coefficients() const;
+    explicit Line(const Vector2 &v1, const Vector2 &v2);
```

Comment:
Explicit is not necessary here
```suggestion
    ine(const Vector2 &v1, const Vector2 &v2);
```